### PR TITLE
Notebook for migrating MSSQL to Bigquery

### PR DESCRIPTION
**Pull request by varunika**
This notebooks provides the below functionalities to the users:

1. Migrate entire database to BQ
2. Migrate specific tables
3. Migrate specific schemas
4. Control parallelism
5. Validate row counts post migration

### CODE REVIEW

**Comment by NiloFreitas** [comment]
I'll review it

**Comment by surjits254** [comment]
Add the solution in main readme and notebooks/readme

File: notebooks/mssql2bq/mssql-to-bigquery-notebook.ipynb [hunk 1]
```diff
@@ -0,0 +1,853 @@
...
+    "import sys\n",
+    "from kfp import dsl\n",
+    "from kfp.v2 import compiler\n",
+    "from datetime import datetime\n",
+    "import time\n",
+    "import copy\n",
+    "import json\n",
+    "import pandas as pd\n",
+    "from google_cloud_pipeline_components.experimental.dataproc import DataprocPySparkBatchOp\n",
+    "import ipywidgets as widgets\n",
```
Comment (surjits254) [inline notebooks/mssql2bq/mssql-to-bigquery-notebook.ipynb:128]: error while importing ipywidgets

File: notebooks/mssql2bq/mssql-to-bigquery-notebook.ipynb [hunk 2]
```diff
@@ -0,0 +1,853 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "af0cf264-6eef-441d-a0a4-0c7bdc3d5b96",
+   "metadata": {},
+   "source": [
+    "# Step 3: Assign Parameters\n",
+    "\n",
+    "## Step 3.1 Common Parameters\n",
```
Comment (surjits254) [inline notebooks/mssql2bq/mssql-to-bigquery-notebook.ipynb:141]: Please add SA in the markdown

File: notebooks/mssql2bq/mssql-to-bigquery-notebook.ipynb [hunk 3]
```diff
@@ -0,0 +1,853 @@
...
+   "id": "ee427d7c-edcb-46b2-a825-624db784aff0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "PROJECT = \"<project-id>\"\n",
+    "REGION = \"<region>\"\n",
+    "GCS_STAGING_LOCATION = \"gs://<staging-bucket>\"\n",
+    "SUBNET = \"projects/<project-id>/regions/<region>/subnetworks/<subnet-name>\"\n",
+    "MAX_PARALLELISM = 5 # default value is set to 5\n",
+    "SERVICE_ACCOUNT = \"\"\n",
```
Comment (surjits254) [inline notebooks/mssql2bq/mssql-to-bigquery-notebook.ipynb:168]: ADD SA as SERVICE_ACCOUNT = "<service-account>"

File: notebooks/mssql2bq/mssql-to-bigquery-notebook.ipynb [hunk 4]
```diff
@@ -0,0 +1,853 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "46e607f2-c6c0-4abe-b4ff-ce34c83fcbf2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mssql_to_bigquery_jobs = []\n",
```
Comment (surjits254) [inline notebooks/mssql2bq/mssql-to-bigquery-notebook.ipynb:552]: can we move this to top in step 3?

**Review by surjits254 — CHANGES_REQUESTED** [review]
minor changes, rest LGTM

**Comment by shashank-google** [comment]
@varunika can you please close the comments on this pull request.

File: notebooks/mssql2bq/mssql-to-bigquery-notebook.ipynb (hunk 3 above)
Comment (varunika) [reply notebooks/mssql2bq/mssql-to-bigquery-notebook.ipynb:168]: SA is already added as SERVICE_ACCOUNT = "" in line 168

File: notebooks/mssql2bq/mssql-to-bigquery-notebook.ipynb (hunk 4 above)
Comment (varunika) [reply notebooks/mssql2bq/mssql-to-bigquery-notebook.ipynb]: Moved this to step 3 where common parameters for the notebooks have been defined.

File: notebooks/mssql2bq/mssql-to-bigquery-notebook.ipynb (hunk 2 above)
Comment (varunika) [reply notebooks/mssql2bq/mssql-to-bigquery-notebook.ipynb:140]: Added SA to markdown

File: notebooks/mssql2bq/mssql-to-bigquery-notebook.ipynb (hunk 1 above)
Comment (varunika) [reply notebooks/mssql2bq/mssql-to-bigquery-notebook.ipynb]: Removed ipywidgets as its not being used anywhere in the notebook.

**Comment by varunika** [comment]
> Add the solution in main readme and notebooks/readme

As for this change, I will create a new pull request after this pull request is merged with main as my dev branch does not have the readme.md file and rebasing the dev branch will destroy the current commit ids present in the PR.

**Review by surjits254 — APPROVED** [review]
LGTM, Please create another commit for readme as well

**Pull request merged**